ddtrace/tracer/spancontext.go

```diff
@@ -164,21 +177,34 @@ func newTrace() *trace {
 	return &trace{spans: make([]*span, 0, traceStartSize)}
 }
 
-func (t *trace) samplingPriority() (p int, ok bool) {
-	t.mu.RLock()
-	defer t.mu.RUnlock()
+// returns the sampling priority of the trace. Not thread safe.
+func (t *trace) samplingPriorityUnsafe() (p int, ok bool) {
```

Comment:
```suggestion
func (t *trace) samplingPriorityLocked() (p int, ok bool) {
```
We use this pattern in multiple places in the repository, and generally the suffix is `*Locked`. Can you please rename it? (it's even used right here as `setSamplingPriorityLocked`)